### PR TITLE
feat(logger): inject custom zap logger into grpc

### PR DESCRIPTION
**Pull request by bsushmith**
making shield relevant changes using https://github.com/odpf/salt/pull/27

### CODE REVIEW

File: cmd/setup.go [hunk 1]
```diff
@@ -44,7 +42,7 @@ func getGRPCMiddleware(cfg *config.Shield, logger log.Logger) grpc.ServerOption
 	return grpc.UnaryInterceptor(
 		grpcMiddleware.ChainUnaryServer(
 			grpc_interceptors.EnrichCtxWithIdentity(cfg.App.IdentityProxyHeader),
-			grpczap.UnaryServerInterceptor(zap.NewExample()),
+			grpczap.UnaryServerInterceptor(logger.(*log.Zap).GetInternalZapLogger().Desugar()),
```
Comment (mabdh) [inline cmd/setup.go:45]: @bsushmith we can make it more defensive by always checking the casting and have a fallback scenario for this

Comment (bsushmith) [reply cmd/setup.go:45]: Then, in that case, would the default fallback be `zap.NewExample()` itself? 🤔

**Comment by krtkvrm** [comment]
@bsushmith any updates on this PR ?

**Comment by bsushmith** [comment]
This PR is no longer needed. The corresponding changes were added and merged through the #164 - https://github.com/odpf/shield/pull/164/files#diff-d0706a4ae4ecf91c5224ca388f31a087a36af06c40dc81321da7958b25c70b23

**Pull request closed without merging**